src/auth/auth.controller.ts

```diff
@@ -1,22 +1,55 @@
-import { Controller, Get, Post, Body, Patch, Param, Delete } from '@nestjs/common';
+import { Controller, Get, Post, Body, Patch, Param, Delete, Response } from '@nestjs/common';
 import { AuthService } from './service/auth.service';
 import { LoginDto } from './dto/login.dto';
 import { RegisterDto } from './dto/register.dto';
-import { ApiTags } from '@nestjs/swagger';
-
+import { ApiResponse, ApiTags } from '@nestjs/swagger';
+import * as express from 'express';
 @ApiTags('auth')
 @Controller('auth')
 export class AuthController {
   constructor(private readonly authService: AuthService) {}
 
   @Post('register')
-    register(@Body() registerDto: RegisterDto) {
-        return this.authService.register(registerDto);
+    async register(@Body() registerDto: RegisterDto) {
+        const response = await this.authService.register(registerDto);
+        await this.authService.createEmailToken(registerDto.email);
+        const emailVerificationResponse = await this.sendEmailVerification(registerDto.email);
```

Comment:
Estas 2 yo creo que si las juntaria 🤔, no estoy seguro
```suggestion
        await this.authService.createEmailToken(registerDto.email);
        const emailVerificationResponse = await this.sendEmailVerification(registerDto.email);
```